packages/settingeditor/src/pluginlist.tsx

```diff
@@ -321,19 +453,34 @@ export class PluginList extends ReactWidget {
             <ul>{modifiedItems}</ul>
           </div>
         )}
-        <h1 className="jp-PluginList-header">{trans.__('Settings')}</h1>
-        <ul>{otherItems}</ul>
+        {otherItems.length > 0 && (
+          <div>
+            <h1 className="jp-PluginList-header">{trans.__('Settings')}</h1>
+            <ul>{otherItems}</ul>
+          </div>
+        )}
+        {modifiedItems.length === 0 && otherItems.length === 0 && (
+          <h1 className="jp-PluginList-noResults">
+            {' '}
+            No items match your search.{' '}
```

Comment:
This needs to be wrapped through `trans.__` to be translated

```suggestion
            {trans.__('No items match your search.)'}
```